leader-election-impl/src/main/java/com/palantir/paxos/PaxosStateLogMigrator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.paxos;
+
+import java.util.stream.LongStream;
+
+import org.immutables.value.Value;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.common.persist.Persistable;
+
+public final class PaxosStateLogMigrator<V extends Persistable & Versionable> {
+    @VisibleForTesting
+    static final int BATCH_SIZE = 10_000;
+
+    private final PaxosStateLog<V> sourceLog;
+    private final PaxosStateLog<V> destinationLog;
+    private final Persistable.Hydrator<V> hydrator;
+    private final SqlitePaxosStateLogMigrationState migrationState;
```

Comment:
Non-actionable: sigh. This was _almost_ entirely agnostic of SQLite. Given we aren't going to be implementing another migrator any time soon though, I'd say leave it.